src/orion/core/utils/terminal.py

```diff
@@ -15,20 +15,36 @@ def ask_question(question, default=None):
         The question to be asked.
     default: str
         The default value to use if the user enters nothing.
+    choice: list
+        List of expected values to check user answer
+    ignore_case: bool
+        Used only if choice is provided. If True, ignore case when checking
+        user answer against given choice.
 
     Returns
     -------
     str
         The answer provided by the user.
 
     """
+    if choice is not None:
+        if ignore_case:
+            choice = [value.lower() for value in choice]
+        question = question + " (choice: {})".format(", ".join(choice))
+
     if default is not None:
         question = question + " (default: {}) ".format(default)
 
-    answer = input(question)
-
-    if answer.strip() == "":
-        return default
+    while True:
+        answer = input(question)
+        if answer.strip() == "":
+            answer = default
+            break
+        if choice is None:
+            break
+        if answer in choice or (ignore_case and answer.lower() in choice):
+            break
+        print("Unexpected value {}\n".format(answer))
```

Comment:
I would add here 'must be one of {}' and then list the valid choices.

Reply:
Done!